platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/bls/addressbook/BlsNodeData.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.platform.bls.addressbook;
+
+import com.swirlds.platform.bls.crypto.BlsPublicKey;
+import com.swirlds.platform.bls.crypto.ShareId;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+/** Class containing data relevant to BLS, for an individual node */
+public class BlsNodeData {
+    /** The number of shares the node has */
+    private int shareCount;
```

Comment:
isn't the share count just the size of the `shareIds` list?

Reply:
I few things are not obvious to me:
- where is this instance stored?
- when is it constructed?
- when are the IDs populated?
- Is it possible to avoid having a list of IDs which might not be populated?

Reply:
These are as-of-yet unanswered questions.

`BlsNodeData`, and the `PlatformAddressBook` more generally, are placeholders. BLS needs this data, but I don't know yet how this will integrate into the existing platform. I created these placeholders to be able to test the logic of the protocols without having to settle on an integration strategy before being ready to do so.